adrs/0001-pseudo-private-custom-properties.md

```diff
@@ -0,0 +1,28 @@
+# Refactor component styling using pseudo-private custom properties
+
+**Date:** [2022-11-03]
+
+### Issue
+Component styles are difficult to maintain. This is due to:
+
+1. Disjointed styles .less files amongst components. Within components, we currently sometime will output contextual, variant, modifier, interaction styles with the base component class and sometimes we don’t. These styles will might be in alphabetical order, by some logical order, in the order added, or seemingly randomly. Many component style .less files include a large preamble and table of contents that are seldom updated to reflect the current state of the component. 
+2. Complex cascade for component styles within variants, modifiers and theme contexts. This makes scanning .less files cumbersome, maintaining styles difficult, and changes liable to cause regressions.
+
+### Decision
+Defining a structure for component styles using pseudo-private custom properties (PPCP) addresses the issues and includes a few other benefits:
```

Comment:
> addresses the issues

So if we add a snippet highlighting the problem in the issue section can we have here the same snippet with PPCPs applied so that the reader can see at a glance the improvement?

Reply:
Good idea! I'll try to work that in.